src/client/opamCommands.ml

```diff
@@ -2424,27 +2429,49 @@ let repository cli =
       let rm = List.filter (fun n -> not (List.mem n names)) in
       let full_wipe = List.mem `All scope in
       let global = global || full_wipe in
-      let gt =
+      let update_gt () =
         OpamRepositoryCommand.update_selection gt
           ~global ~switches:switches rm
       in
       if full_wipe then
+        let gt = update_gt () in
         OpamRepositoryState.with_ `Lock_write gt @@ fun rt ->
-        check_for_repos rt names
+        let repos =
+          OpamRepositoryName.Map.keys rt.OpamStateTypes.repositories
+        in
+        check_for_repos repos names
```

Comment:
Instead of creating a function that just does a call to ignore it, isn't it better to ignore directly the call to the one returning a boolean?
```suggestion
        ignore @@ check_for_repos' repos names

Reply:
done